src/ProviderIntegrationTest.php

```diff
@@ -228,7 +230,10 @@ public function testExceptions($query, string $exceptionClass, ResponseInterface
         }
     }
 
-    public function exceptionDataProvider()
+    /**
+     * @return array<array{GeocodeQuery|ReverseQuery, class-string<Exception>, Response, string}>
+     */
+    public function exceptionDataProvider(): array
```

Comment:
This should be refactored for phpunit v10 as it allows only static data providers

Reply:
Do you think we should already upgrade to PHPUnit 10 or use a separate PR ?

Reply:
I think separate would be better if there will be more problems to fix